src/helpers.rs

```diff
@@ -793,3 +793,12 @@ pub fn get_local_crates(tcx: &TyCtxt<'_>) -> Vec<CrateNum> {
     }
     local_crates
 }
+
+/// Formats an AllocRange like [0x1..0x3], for use in diagnostics.
+pub struct HexRange(pub AllocRange);
```

Comment:
We should have this somewhere in rustc, I am printing AllocRange in debug messages there and this would be much nicer.

(Not a blocker, of course.)